### DISCUSSION TITLE
[BUG] [INSTALL] HTTP ERROR 500

### DISCUSSION THREAD

**Discussion by globule** [Leantime/leantime#1106] [General] [answered]
Apache2 PHP 8.1.12 / 10.3.34-MariaDB-0ubuntu0.20.04.1

**To Reproduce**
1. Leantime v2.2.10 unziped in /var/www/html/leantime
2. composer install OK
3. npm install OK
4. /etc/apache2/sites-available/leantime.conf                                                                                               
```
<VirtualHost *:80>
        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/html/leantime/public
        ServerAlias leantime.localhost

        <Directory /var/www/html/leantime/public>
                Options Indexes FollowSymLinks
                AllowOverride All
                Require all granted
        </Directory>
</VirtualHost>
```
5. DB created with credentials
```
$mysql -u leantime -p "leantime"
Enter password: 
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MariaDB connection id is 245
```
6. configuration.php
```
   public $dbHost = 'localhost';                         //Database host
    public $dbUser = 'leantime';                          //Database username
    public $dbPassword = 'xxxxxxxxx';              //Database password
    public $dbDatabase = 'leantime';                      //Database name
    public $dbPort = '3306';                              //Database port
```
Tested with a single php file using PDO ->OK

7. Errors :
http://leantime.localhost/ HTTP ERROR 500
http://leantime.localhost/install HTTP ERROR 500

8. files credentials
`chmod -R 777 /var/www/html/leantime`
same problem

.htaccess
```
RewriteEngine On

#Uncomment and update if Leantime runs in a subfolder
#RewriteBase /leantime/public/

RewriteRule ^/?$ index.php?act=dashboard.show
RewriteRule ^([^/\.]+)/?$ index.php?act=$1 [QSA]
RewriteRule ^([^/\.]+)/([^/\.]+)/?$ index.php?act=$1.$2 [QSA]
RewriteRule ^([^/\.]+)/([^/\.]+)/([^/\.]+)/?$ index.php?act=$1.$2&id=$3 [QSA]
```
Leantime error.log:
`[05-Nov-2022 14:06:50 Europe/Paris] PHP Fatal error:  Uncaught PDOException: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'leantime.zp_settings' doesn't exist in /var/www/html/leantime/src/domain/setting/repositories/class.setting.php:42`

test.php file in /public
```
<?php
    error_reporting(1);
    $dbtype = 'mysql';
    $host = 'localhost';
    $user = 'leantime';
    $pass = 'xxxxxxxxxx';
    $database = 'leantime';
    $dsn = $dbtype . ':dbname=' . $database . ";host=" . $host;
    $db = new pdo($dsn, $user, $pass);
    $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_WARNING);
    $db->setAttribute(PDO::ATTR_TIMEOUT, "0");
       // $db->exec($option);
    $stm = $db->query( 'SHOW FULL TABLES' );
    $rows = $stm->fetchAll();
    print_r($rows);
```
Result
`Array ( [0] => Array ( [Tables_in_leantime] => test [0] => test [Table_type] => BASE TABLE [1] => BASE TABLE ) )`

**Comment by globule** [comment, marked as answer]
I found a solution by modifying a method in src/domain/setting/repositories/class.setting.php : 
```
        public function checkIfInstalled()
        {

            try {

                $stmn = $this->db->database->prepare("SHOW TABLES LIKE 'zp_user'");

                $stmn->execute();
                $values = $stmn->fetchAll();
                $stmn->closeCursor();

                if( !count( $values ))
                    return false;

                $stmn = $this->db->database->prepare("SELECT COUNT(*) FROM zp_user");

                $stmn->execute();
                $values = $stmn->fetchAll();
                $stmn->closeCursor();

                return true;

            } catch (PDOException $e) {

                return false;

            }
        }

```

**Reply by marcelfolaron** [reply]
Thank you for this. I am wondering though why it didn't fail the first time. It must assume that zp_user is somewhere there. Can you create a Pull Request with that change?

**Reply by globule** [reply]
I tried to restart from scratch and some other problems appeared, suggesting an update instead of install.
`$_SESSION['isInstalled'] == true`
It was a coockie problem !

Anyway, CSS are not loaded, I'll take a look at the forum to correct all then I'll run a diff and submit it

**Comment by globule** [comment]
Everything solved : https://github.com/Leantime/leantime/issues/401
